samples-final/AspNetCore-EchoBot-With-State/Pages/Index.cshtml

```diff
@@ -0,0 +1,54 @@
+﻿@page
+
+@using Microsoft.Extensions.Options;
+@using  AspNetCore_EchoBot_With_State;
+
+@inject IOptions<ApplicationConfiguration> OptionsApplicationConfiguration;
+
+<!DOCTYPE html>
+<html>
+<head>
+    <title></title>
+    <meta charset="utf-8" />
+
+</head>
+<body style="font-family:'Segoe UI'">
+    <h1>EchoBot sample using ASP.Net Core 2</h1>
+    <p>Describe your bot here and your terms of use etc.</p>
+    <p>To debug you bot using the Bot Framework Emulator, paste this URL into the Emulator window:</p>
+    <div style="padding-left: 50px;">
+        <a id="bot-url"></a>
+    </div>
+    <p>Or click the following link to debug your bot using the <a href="https://github.com/Microsoft/BotFramework-Emulator/releases" target="_blank">Bot Framework Emulator v4</a>:</p>
+    <div style="padding-left: 50px;">
+        <a id="emulator-link"></a>
+    </div>
+    <p>Or open a live chat with your bot in the Bot Framework Emulator!</p>
+    <p>Visit <a href="https://www.botframework.com/">Bot Framework</a> to register your bot. When you register, remember to set the endpoint to:</p>
+    <div style="padding-left: 50px;">
+        https://<strong>{your domain name}</strong>/api/messages
+    </div>
+    <input id="msa-app-id" type="hidden" value="@OptionsApplicationConfiguration.Value.MicrosoftAppId" />
+    <input id="msa-app-pw" type="hidden" value="@OptionsApplicationConfiguration.Value.MicrosoftAppPassword" />
+    <script>
```

Comment:
If you've got yourself a full blow Razor page now, I think it might make more sense to just emit the correct link from the server side rendering template rather than emitting two hidden inputs and then running additional client side JS to emit the link. Just keeps it way simpler IMHO.

Reply:
Ok awesome, let me look into that.